commons-web/src/main/java/eu/europeana/api/commons/web/controller/BaseRestController.java

```diff
@@ -38,4 +39,27 @@ public void verifyReadAccess(HttpServletRequest request)
 	     throws ApplicationAuthenticationException{
 	 getAuthorizationService().authorizeReadAccess(request); 	
    }
+    
+    /**
+     * This method compares If-Match header with the current etag value.
+     * 
+     * @param etag    The current etag value
+     * @param request The request containing If-Match header
+     * @throws HttpException
+     */
+    public void checkIfMatchHeader(String etag, HttpServletRequest request) throws HttpException {
+    	getAuthorizationService().checkIfMatchHeader(etag, request);
+    }
+	
+    /**
+     * This method generates etag for response header.
+     * 
+     * @param timestamp The date of the last modification
+     * @param format       The MIME format
+     * @param version      The API version
+     * @return etag value
+     */
+    public String generateETag(Date timestamp, String format, String version) {
+		return getAuthorizationService().generateETag(timestamp, format, version);
```

Comment:
the implementation must be here not delegated to authorizationService

Reply:
implemented in BaseRestController